utils/install-dependencies.sh

```diff
@@ -45,31 +45,25 @@ function install_dependencies_with_aur() {
 # Install dependencies on centos and fedora
 function install_dependencies_with_yum() {
     sudo yum install -y yum-utils
-
-    local common_dep="curl wget git gcc openresty-openssl111-devel unzip pcre pcre-devel openldap-devel"
-    if [ "${1}" == "centos" ]; then
-        # add APISIX source
-        local apisix_pkg=apache-apisix-repo-1.0-1.noarch
-        rpm -q --quiet ${apisix_pkg} || sudo yum install -y https://repos.apiseven.com/packages/centos/${apisix_pkg}.rpm
-
-        # install apisix-runtime and some compilation tools
-        # shellcheck disable=SC2086
-        sudo yum install -y apisix-runtime $common_dep
-    else
-        # add OpenResty source
-        sudo yum-config-manager --add-repo "https://openresty.org/package/${1}/openresty.repo"
-
-        # install OpenResty and some compilation tools
-        # shellcheck disable=SC2086
-        sudo yum install -y openresty $common_dep
+    sudo yum-config-manager --add-repo "https://openresty.org/package/${1}/openresty.repo"
+    if [[ "${1}" == "centos" ]]; then
+        sudo yum -y install centos-release-scl
+        sudo yum -y install devtoolset-9 patch wget
+        set +eu
+        source scl_source enable devtoolset-9
+        set -eu
```

Comment:
What do these dependencies do? There was no such thing before

Reply:
Previously, we used to use compiled apisix runtime repo packages, but now building from source code requires these dependencies.
Centos requires a higher version of gcc..

Reply:
what's the reason to compile apisix-runtime from the source?

Reply:
1. Publishing the apisix-runtime repo will cause the installation of the apisix 3.7.0 debian package to fail.
2. CI is built using source code and should be consistent.
3. The apisix deb/rpm is a complete package and does not rely on the apisix runtime repo package, so building the apisix runtime is just an intermediate process.